rust/noosphere-storage/src/ephemeral.rs

```diff
@@ -0,0 +1,237 @@
+use std::sync::Arc;
+
+use crate::{KeyValueStore, Store};
+use anyhow::{anyhow, Result};
+use async_trait::async_trait;
+use futures::Future;
+use noosphere_common::ConditionalSync;
+use tokio::sync::{Mutex, OnceCell};
+
+/// Provides an [EphemeralStore] that does not persist after dropping.
+#[cfg_attr(not(target_arch = "wasm32"), async_trait)]
+#[cfg_attr(target_arch = "wasm32", async_trait(?Send))]
+pub trait EphemeralStorage: ConditionalSync {
+    type EphemeralStoreType: KeyValueStore + Disposable;
```

Comment:
need better name for `EphemeralStoreType`, as the return value is `EphemeralStore<Self::EphemeralStoreType>` -- `EphemeralStoreInner`?